### PR TITLE
Deprecate data utils

### PR DESCRIPTION
**Pull request by harishsdev**
Deprecate data utils

PR for https://github.com/pytorch/audio/issues/1883

### CODE REVIEW

File: torchaudio/datasets/cmuarctic.py [hunk 1]
```diff
@@ -148,7 +148,7 @@ def __init__(self,
             if not os.path.isdir(self._path):
                 if not os.path.isfile(archive):
                     checksum = _CHECKSUMS.get(url, None)
-                    download_url(url, root, hash_value=checksum, hash_type="md5")
+                    download_url_to_file(url, root, hash_prefix=checksum)
```
Comment (mthrok) [inline torchaudio/datasets/cmuarctic.py:151]: `download_url_to_file` can only handle SHA256 hash, therefore the MD5 sum cannot be passed to the function. 
Either the checksum needs to be updated to SHA256 or validation has to be performed separately after the download.

File: torchaudio/datasets/utils.py [hunk 2]
```diff
@@ -52,67 +52,6 @@ def stream_url(url: str,
             pbar.update(len(chunk))
 
 
-def download_url(url: str,
```
Comment (mthrok) [inline torchaudio/datasets/utils.py:55]: We cannot delete the function until the next release. The `deprecated` decorator has to be put on this `download_url`.

File: torchaudio/datasets/yesno.py [hunk 3]
```diff
@@ -44,6 +44,11 @@ def __init__(
 
         self._parse_filesystem(root, url, folder_in_archive, download)
 
+
+@_mod_utils.deprecated(
+    "donwload_url_to_file will be dericated in next version that is v0.12"
+    "version=0.12"
+)
```
Comment (mthrok) [inline torchaudio/datasets/yesno.py:51]: This decorator should be added on the implementation of `download_url_to_file`.
Also, `version` has to be keyword argument, not string.

**Review by mthrok — CHANGES_REQUESTED** [review]
Hi @harishsdev 

Thanks for working on this. Please refer to my comments.

**Comment by krishnakalyan3** [comment]
@harishsdev please let me know if you will continue to work on this issue. Else I would want to take a stab at this.

**Comment by harishsdev** [comment]
@krishnakalyan3 please continue,im no more working on this isssue

**Pull request closed without merging**